### PR TITLE
Add environment Manager to handle auto and user env variables

### PR DESCRIPTION
**Pull request by isseis**
This pull request implements the core logic and tests for managing environment variables in the command runner. It introduces a new `Manager` interface and its implementation for handling both user-defined and auto-generated environment variables, along with comprehensive unit tests to ensure correct validation and merging behavior.

**Environment Variable Management Implementation:**

* Added a new `Manager` interface in `internal/runner/environment/manager.go` to manage environment variables, including methods for validating user environment variable names and building the final merged environment.
* Implemented the `manager` struct and its methods, including validation to prevent user variables from using reserved prefixes and logic to merge user and auto-generated variables.

**Testing and Validation:**

* Created comprehensive tests in `internal/runner/environment/manager_test.go` covering validation of reserved prefixes, correct merging of user and auto-generated variables, and handling of edge cases such as empty inputs and default clock usage.

**Documentation and Planning Updates:**

* Updated the implementation plan in `docs/tasks/0028_automatic_env_vars/04_implementation_plan.md` to reflect completion of the interface definition, implementation, and testing phases, and to align naming with linting requirements (renaming `EnvironmentManager` to `Manager`).

### CODE REVIEW

File: internal/runner/environment/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/isseis/go-safe-cmd-runner/internal/runner/runnertypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestManagerValidateUserEnvNames(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/environment/manager_test.go:13]: WAI
